tests/test_fields_bytes.py

```diff
@@ -71,5 +71,23 @@ class Foo(proto.Message):
     # for strings (but not vice versa).
     foo.bar = b"anything"
     assert foo.bar == "anything"
-    with pytest.raises(TypeError):
-        foo.baz = "anything"
+
+    # We need to permit setting bytes fields from strings,
+    # but the marshalling needs to base64 decode the result.
+    # This is a requirement for interop with the vanilla protobuf runtime:
+    # converting a proto message to a dict base64 encodes the bytes
+    # becase it may be sent over the network via a protocol like HTTP.
+    foo.baz = "dW5sYWRlbiBzd2FsbG93"
+    assert foo.baz == b"unladen swallow"
```

Comment:
Using the encoded literal is like having "magic numbers" in tests.  Please encode the value from a constant, e.g.:
```python
    unladen_swallow = b"unladen swallow"
    encoded_swallow = base64.urlsafe_b64encode(unladen_swallow)
    foo.baz = encoded_swallow
    assert foo.baz == unladen_swallow
```

Reply:
As with my other comment, this test as-is is much easier to verify than the proposed.

Reply:
Really?  You can eyeball that encoded string and see that it should decode to `b'unladen swallow'`?